mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipArchivesLoaderTest.java

```diff
@@ -0,0 +1,145 @@
+package org.apache.james.mailbox.backup;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.backup.zip.ZipArchivesLoader;
+import org.apache.james.mailbox.backup.zip.Zipper;
+import org.apache.james.mailbox.inmemory.manager.InMemoryIntegrationResources;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.github.fge.lambdas.Throwing;
+
+public class ZipArchivesLoaderTest implements MailboxMessageFixture {
+    private static final int BUFFER_SIZE = 4096;
+
+    private final ArchiveService archiveService = new Zipper();
+    private final MailArchivesLoader archiveLoader = new ZipArchivesLoader();
+
+    private MailArchiveRestorer archiveRestorer;
+    private MailboxManager mailboxManager;
+    private DefaultMailboxBackup backup;
+
+    @BeforeEach
+    void beforeEach() {
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        archiveRestorer = new ZipMailArchiveRestorer(mailboxManager, archiveLoader);
+        backup = new DefaultMailboxBackup(mailboxManager, archiveService, archiveRestorer);
+    }
+
+    private void createMailBoxWithMessage(MailboxPath mailboxPath, MailboxMessage... messages) throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(mailboxPath.getUser());
+        MailboxId mailboxId = mailboxManager.createMailbox(mailboxPath, session).get();
+        Arrays.stream(messages).forEach(Throwing.consumer(message ->
+            {
+                MessageManager.AppendCommand appendCommand = MessageManager.AppendCommand.builder()
+                    .withFlags(message.createFlags())
+                    .build(message.getFullContent());
+                mailboxManager.getMailbox(mailboxId, session).appendMessage(appendCommand, session);
+            }
+            )
+        );
+    }
+
+    @Test
+    void mailAccountIteratorFromEmptyArchiveShouldBeEmpty() throws Exception {
+        ByteArrayOutputStream destination = new ByteArrayOutputStream(BUFFER_SIZE);
+        backup.backupAccount(USER1, destination);
+
+        InputStream source = new ByteArrayInputStream(destination.toByteArray());
+        MailArchiveIterator mailArchiveIterator = archiveLoader.load(source);
+
+        assertThat(mailArchiveIterator.hasNext()).isEqualTo(false);
+        assertThat(mailArchiveIterator.next()).isEqualTo(null);
+    }
+
+    @Test
+    void callingNextSeveralTimeOnAnEmptyIteratorShouldReturnNull() throws Exception {
```

Comment:
it should not return IOException?

Reply:
backup.backupAccount throws MailboxException too.

Reply:
I was talking about next(): when you call it multiple times, should it return null or throw?

Reply:
Ah ok, i thought null was adequate but it seems to exist a rule to throw a NoSuchElementException

https://rules.sonarsource.com/java/tag/unpredictable/RSPEC-2272